k8s/k8s.go

```diff
@@ -313,46 +305,41 @@ func CmdAddK8s(ctx context.Context, args *skel.CmdArgs, conf types.NetConf, epID
+// pod name and namespace rather than container ID, so we may receive multiple DEL calls for the same pod, but with different container IDs.
+// As such, we must only delete the workload endpoint when the provided CNI_CONATAINERID matches the value on the WorkloadEndpoint. If they do not match,
+// it means the DEL is for an old sandbox and the pod is still running. We should still clean up IPAM allocations, since they are identified by the
+// container ID rather than the pod name and namespace. If they do match, then we can delete the workload endpoint.
 func CmdDelK8s(ctx context.Context, c calicoclient.Interface, epIDs utils.WEPIdentifiers, args *skel.CmdArgs, conf types.NetConf, logger *logrus.Entry) error {
 	wep, err := c.WorkloadEndpoints().Get(ctx, epIDs.Namespace, epIDs.WEPName, options.GetOptions{})
 	if err != nil {
-		if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
-			// We can talk to the datastore but WEP doesn't exist in there,
-			// but we still want to go ahead with the clean up. So, log a warning and continue with the clean up.
-			logger.WithField("WorkloadEndpoint", epIDs.WEPName).Warning("WorkloadEndpoint does not exist in the datastore, moving forward with the clean up")
-		} else {
+		if _, ok := err.(cerrors.ErrorResourceDoesNotExist); !ok {
 			// Could not connect to datastore (connection refused, unauthorized, etc.)
-			// so we have no way of knowing/checking ActiveInstanceID. To protect the endpoint
+			// so we have no way of knowing/checking ContainerID. To protect the endpoint
 			// from false DEL, we return the error without deleting/cleaning up.
 			return err
 		}
 
-		// Check if ActiveInstanceID is populated (it will be an empty string "" if it was populated
-		// before this field was added to the API), and if it is there then compare it with ContainerID
-		// passed by the orchestrator to make sure they are the same, return without deleting if they aren't.
+		// The WorkloadEndpoint doesn't exist for some reason. We should still try to clean up any IPAM allocations
+		// if they exist, so continue DEL processing.
+		logger.WithField("WorkloadEndpoint", epIDs.WEPName).Warning("WorkloadEndpoint does not exist in the datastore, moving forward with the clean up")
 	} else if wep.Spec.ContainerID != "" && args.ContainerID != wep.Spec.ContainerID {
-		logger.WithField("WorkloadEndpoint", wep).Warning("CNI_CONTAINERID does not match WorkloadEndpoint ActiveInstanceID so ignoring the DELETE cmd.")
-		return nil
-
-		// Delete the WorkloadEndpoint object from the datastore.
-		// In case of k8s, where we are deleting the WEP we got from the Datastore,
-		// this Delete is a Compare-and-Delete, so if *any* field in the WEP changed from
-		// the time we get WEP until here then the Delete operation will fail.
+		// If the ContainerID is populated and doesn't match the CNI_CONATINERID provided for this execution, then
```

Comment:
lol I was waiting to see when you'd move the comments inside the `if...` block :trollface: